integration-tests/src/test/java/org/zowe/apiml/integration/ha/WebSocketChaoticTest.java

```diff
@@ -0,0 +1,160 @@
+
+    @BeforeAll
+    static void setup() {
+        String plainCred = "user:pass";
+        String base64cred = Base64.getEncoder().encodeToString(plainCred.getBytes());
+        VALID_AUTH_HEADERS.add("Authorization", "Basic " + base64cred);
+    }
+
+    private TextWebSocketHandler appendResponseHandler(StringBuilder target, int countToNotify) {
+        final AtomicInteger counter = new AtomicInteger(countToNotify);
+        return new TextWebSocketHandler() {
+            @Override
+            public void handleTextMessage(WebSocketSession session, TextMessage message) {
+                synchronized (target) {
+                    target.append(message.getPayload());
+                    if (counter.decrementAndGet() == 0) {
+                        target.notify();
+                    }
+                }
+            }
+
+            @Override
+            public void afterConnectionClosed(WebSocketSession session, CloseStatus status) {
+                synchronized (target) {
+                    target.append(status.toString());
+                    if (counter.decrementAndGet() == 0) {
+                        target.notify();
+                    }
+                }
+            }
+        };
+    }
+
+    private String discoverableClientGatewayUrl(String gatewayUrl, int index) throws URISyntaxException {
+        return haGatewayRequests.gatewayServices.get(index).getGatewayUriWithPath(gatewayUrl).toString();
+    }
+
+    private WebSocketSession appendingWebSocketSession(String url, WebSocketHttpHeaders headers, StringBuilder response, int countToNotify)
+        throws Exception {
+        StandardWebSocketClient client = new StandardWebSocketClient();
+        client.getUserProperties().put(SSL_CONTEXT_PROPERTY, HttpClientUtils.ignoreSslContext());
+        URI uri = UriComponentsBuilder.fromUriString(url).build().encode().toUri();
+        return client.doHandshake(appendResponseHandler(response, countToNotify), headers, uri).get(30000, TimeUnit.MILLISECONDS);
+    }
+
+    @Nested
+    class WhenRoutingSessionGivenHA {
+        @Nested
+        class OpeningASession {
+            @Nested
+            class WhenAnInstanceIsOff {
+                @ParameterizedTest(name = "WhenRoutingSessionGivenHA.OpeningASession.WhenAnInstanceIsOff#propagateTheSessionToTheAliveInstance {0}")
+                @ValueSource(strings = {"/discoverableclient/ws/v1/uppercase", "/ws/v1/discoverableclient/uppercase"})
+                void propagateTheSessionToTheAliveInstance(String path) throws Exception {
+                    final StringBuilder response = new StringBuilder();
+
+                    WebSocketSession session = appendingWebSocketSession(discoverableClientGatewayUrl(path, 0), VALID_AUTH_HEADERS, response, 1);
+
+                    // shutdown one instance of DC to check whether the message can reach out the other instance
+                    haDiscoverableClientRequests.shutdown(0);
```

Comment:
Doesn't it break the tests if this happens before the next one? 
Or do you run this test separately?

Reply:
Now that I checked in the gh actions, `newSessionCanBeCreated` got executed before this test. I will change the execution order to see what happens

Reply:
I've defined the test execution order so that this test runs before the others, and actually no tests are failing